lib/exports.js

```diff
@@ -4,9 +4,45 @@ var util = require('util');
 var swaggerRouter = require('swagger-router');
 var yaml = require('js-yaml');
 var fs = require('fs');
+var EventEmitter = require('events').EventEmitter;
 
 var exports = {};
 
+function Lifecycle() {
+    EventEmitter.call(this);
+    this._initialized = false;
+}
+util.inherits(Lifecycle, EventEmitter);
+
+/**
+ * Lazily sets up forwarding of the application-lifecycle events
+ * to the exported object. Only called if the client subscribed to
+ * some events.
+ *
+ * @private
+ */
+Lifecycle.prototype._setup = function() {
+    var self = this;
+    // The server will certainly be initialized by now
+    var server = require('./server').server;
+    server.on('close', function() { self.emit('close'); });
+};
+[
+    'addListener',
+    'on',
+    'once',
+    'prependListener',
+    'prependOnceListener'
```

Comment:
@d00rman `Lifecycle` inherits `removeListener` from the `EventEmitter`. We only need to override methods that add listeners to set up lazy initialisation of the event forwarding. Other methods don't need modification, so there inherited as is.